# OBS001 — Local CO Environment of ASPECS C19

## Scientific Objective

Determine whether ASPECS C19 is isolated or resides within a physically
associated galaxy environment by analysing nearby ASPECS CO detections.

### Target

ASPECS C19

RA (J2000): 03:32:36.19

Dec (J2000): −27:46:28.0

Adopted Redshift:

z = 2.574

### Products

- Neighbor catalog
- Angular separations
- Redshift comparison
- Projected separations
- Environment map
- Scientific summary

In [1]:
# ==========================================================
# Phase 1A
# Imports
# ==========================================================

import pandas as pd
import numpy as np

from pathlib import Path

from astropy.table import Table
from astropy.coordinates import SkyCoord
import astropy.units as u

from astropy.cosmology import Planck18

import matplotlib.pyplot as plt

In [2]:
# ==========================================================
# Phase 1A
# Project Paths
# ==========================================================

PROJECT = Path.cwd().parent

DATA = PROJECT / "data"
CATALOGS = DATA / "catalogs"

print(CATALOGS)

/home/glenn/Projects/CosmicIntelligenceLab/data/catalogs


In [3]:
# ==========================================================
# Phase 1A
# Target Definition
# ==========================================================

c19 = SkyCoord(
    "03h32m36.19s",
    "-27d46m28.0s",
    frame="icrs"
)

c19_redshift = 2.574

In [4]:
# ==========================================================
# Phase 1A
# Inventory Available Catalogs
# ==========================================================

for f in sorted(CATALOGS.glob("*")):
    print(f.name)

.gitkeep
aspecs_co_catalog.ecsv
clean_source_catalog.ecsv
clustered_galaxies.ecsv
einstein_master_catalog.ecsv
hudf_search.ecsv
ml_features.ecsv
object39_comparison_catalog.ecsv
object39_ned.ecsv
object39_profile.ecsv
object45_profile.ecsv
obs001_co_neighbor_analysis.ecsv
source_catalog.ecsv


In [5]:
# ==========================================================
# Phase 1A
# Inspect Einstein Master Catalog
# ==========================================================

master = Table.read(CATALOGS / "einstein_master_catalog.ecsv")

print(master)

master.colnames
master[:5]

label     x_centroid     ...       ra_deg             dec_deg      
                         ...                                       
----- ------------------ ... ------------------ -------------------
    4 121.18956760830898 ... 53.135765607444675  -27.78501220087869
    5  529.4126604123757 ...  53.14787352892725 -27.774151044328633
    6  540.0202264635453 ... 53.148144506746554  -27.77383747084123
    7  809.1891942549092 ...  53.15541744273373  -27.76616059899571
    8   620.784680169983 ...  53.15060092496098 -27.771726351899147
    9   410.067776025836 ... 53.145255102995804 -27.777985531753448
   10  678.6195422845603 ...  53.15229613377274  -27.77016884527462
   11 375.53805295333785 ...  53.14451370793668 -27.779107503914037
   12  542.4486261402996 ... 53.149088191189925 -27.774387277318727
  ...                ... ...                ...                 ...
   75 263.78269123508545 ...  53.16302038759664 -27.797713163770144
   76  850.5606247989522 ...  53.17908332385395 

label,x_centroid,y_centroid,sky_centroid,bbox_xmin,bbox_xmax,bbox_ymin,bbox_ymax,area,semimajor_axis,semiminor_axis,orientation,eccentricity,min_value,max_value,segment_flux,segment_flux_err,kron_flux,kron_flux_err,cluster,ra_deg,dec_deg
,,,,,,,,pix2,pix,pix,deg,,,,,,,,,,
int32,float64,float64,object,int64,int64,int64,int64,float64,float64,float64,float64,float64,float64,float64,float64,float64,float64,float64,int32,float64,float64
4,121.18956760830898,31.32029442469728,None,120,122,30,32,6.0,0.6440643877584695,0.5514238892637224,294.4995327788689,0.516706370668758,0.5479127168655396,2.140134334564209,6.465743601322174,nan,12.983454270407863,nan,0,53.135765607444675,-27.78501220087869
5,529.4126604123757,79.48980618510005,None,525,532,76,84,32.0,2.2941649471403283,1.4005674373639603,303.90657994640236,0.7920230641969821,0.5414181351661682,1.639194130897522,24.018874406814575,nan,95.29540050691291,nan,2,53.14787352892725,-27.774151044328633
6,540.0202264635453,79.18836128348309,None,538,542,77,81,15.0,1.063488324717736,0.9223128857493856,271.8012506450832,0.497868595189019,0.5437135696411133,1.8551239967346191,12.340121269226074,nan,60.14385696013455,nan,2,53.148144506746554,-27.77383747084123
7,809.1891942549092,84.74331593280078,None,808,811,83,86,13.0,1.0771237638461464,0.7989851718607586,45.38350344446989,0.6706470897789912,0.5706205368041992,1.5475101470947266,11.399610340595245,nan,17.897842162107004,nan,2,53.15541744273373,-27.76616059899571
8,620.784680169983,90.4208343944767,None,619,622,89,92,10.0,1.13510003434925,0.6721785777757441,318.5683560480159,0.805808924882241,0.5459803342819214,0.6996515989303589,6.043212354183197,nan,46.91598378461046,nan,0,53.15060092496098,-27.771726351899147


In [6]:
# ==========================================================
# Phase 1A
# Inspect Object 39 Catalog
# ==========================================================

object39 = Table.read(
    CATALOGS / "object39_profile.ecsv"
)

print(object39)

object39.colnames

Einstein Label        SIMBAD Name         ...      SIMBAD Dec    
-------------- -------------------------- ... -------------------
            39 ATCDFS J033238.75-274632.5 ... -27.775694444444447


['Einstein Label',
 'SIMBAD Name',
 'Object Type',
 'Measured Flux',
 'Area (pixels)',
 'Semimajor Axis (pixels)',
 'Semiminor Axis (pixels)',
 'Orientation (deg)',
 'Eccentricity',
 'Einstein RA (deg)',
 'Einstein Dec (deg)',
 'SIMBAD RA',
 'SIMBAD Dec']

In [7]:
# ==========================================================
# Phase 1A
# Inspect Object 39 Comparison Catalog
# ==========================================================

comparison = Table.read(
    CATALOGS / "object39_comparison_catalog.ecsv"
)

print(comparison)

comparison.colnames

 Seq   RAJ2000   DEJ2000   best_z zspec ... AGN Ksmag  IR36mag IR45mag
         deg       deg                  ...      mag     mag     mag  
----- --------- ---------- ------ ----- ... --- ------ ------- -------
10989 53.160816 -27.775379   0.64    -- ...   0 20.489  20.838  21.232
10645 53.161629 -27.780247  0.619 0.619 ...   0 19.964  20.638  21.002
10646 53.158283 -27.780985   0.59    -- ...   0 23.842  24.184  24.771
10587 53.158138 -27.781094   0.64    -- ...   0 21.563  22.083  22.362
11316 53.168579 -27.772926   0.61    -- ...   0 25.055  26.199  26.016
11688 53.158356 -27.768261   0.63    -- ...   0 26.093  25.279  25.962
11309 53.151211 -27.772839    0.6    -- ...   0 23.652  24.066  24.096
11209  53.16993  -27.77103   0.67    -- ...   0 19.292    19.5  19.952


['Seq',
 'RAJ2000',
 'DEJ2000',
 'best_z',
 'zspec',
 'zphot',
 'sep_arcsec',
 'AGN',
 'Ksmag',
 'IR36mag',
 'IR45mag']

## Phase 1A Findings

The Object 39 comparison catalog provides:

- Accurate sky coordinates
- Spectroscopic and photometric redshifts
- Angular separations from Object 39
- AGN classifications
- Near-infrared and Spitzer photometry

The catalog does **not** contain:

- ASPECS CO identifiers
- CO transition information
- CO luminosities
- Molecular gas masses

Therefore, this catalog cannot by itself answer the primary questions of OBS001.

In [8]:
from astropy.table import Table

co_sources = [
    {
        "id": 1,
        "ra": "03:32:38.54",
        "dec": "-27:46:34.62",
        "z": 2.543,
        "jup": 3,
        "snr": 37.7,
    },

    {
        "id": 2,
        "ra": "03:32:42.38",
        "dec": "-27:47:07.92",
        "z": 1.317,
        "jup": 2,
        "snr": 17.9,
    },

    {
        "id": "3",
        "ra": "03:32:41.02",
        "dec": "-27:46:31.56",
        "z": 2.453,
        "jup": 3,
        "snr": 15.8,
    },

    {
        "id": "4",
        "ra": "03:32:34.44",
        "dec": "-27:46:59.82",
        "z": 1.414,
        "jup": 2,
        "snr": 15.5,
    },

    {
        "id": "5",
        "ra": "03:32:39.76",
        "dec": "-27:46:11.58",
        "z": 1.550,
        "jup": 2,
        "snr": 15.0,
    },

    {
        "id": "6",
        "ra": "03:32:39.90",
        "dec": "-27:47:15.12",
        "z": 1.095,
        "jup": 2,
        "snr": 11.9,
    },

    {
        "id": "7",
        "ra": "03:32:43.53",
        "dec": "-27:46:39.47",
        "z": 2.697,
        "jup": 3,
        "snr": 10.9,
    },

    {
        "id": "8",
        "ra": "03:32:35.58",
        "dec": "-27:46:26.16",
        "z": 1.382,
        "jup": 2,
        "snr": 9.5,
    },

    {
        "id": "9",
        "ra": "03:32:44.03",
        "dec": "-27:46:36.05",
        "z": 2.698,
        "jup": 3,
        "snr": 9.3,
    },

    {
        "id": "10",
        "ra": "03:32:42.98",
        "dec": "-27:46:50.45",
        "z": 1.037,
        "jup": 2,
        "snr": 8.7,
    },

    {
        "id": "11",
        "ra": "03:32:39.80",
        "dec": "-27:46:53.70",
        "z": 1.096,
        "jup": 2,
        "snr": 7.9,
    },

    {
        "id": "12",
        "ra": "03:32:36.21",
        "dec": "-27:46:27.78",
        "z": 2.574,
        "jup": 3,
        "snr": 7.0,
    },

    {
        "id": "13",
        "ra": "03:32:35.56",
        "dec": "-27:47:04.32",
        "z": 3.601,
        "jup": 4,
        "snr": 6.8,
    },

    {
        "id": "14",
        "ra": "03:32:34.84",
        "dec": "-27:46:40.74",
        "z": 1.098,
        "jup": 2,
        "snr": 6.7,
    },

    {
        "id": "15",
        "ra": "03:32:36.48",
        "dec": "-27:46:31.92",
        "z": 1.096,
        "jup": 2,
        "snr": 6.5,
    },

    {
        "id": "16",
        "ra": "03:32:39.92",
        "dec": "-27:46:07.44",
        "z": 1.294,
        "jup": 2,
        "snr": 6.4,
    },

    {
        "id": "MP01",
        "ra": "03:32:37.30",
        "dec": "-27:45:57.80",
        "z": 1.096,
        "jup": 2,
        "snr": 4.5,
    },

    {
        "id": "MP02",
        "ra": "03:32:35.48",
        "dec": "-27:46:26.50",
        "z": 1.087,
        "jup": 2,
        "snr": 4.0,
    },
]

In [9]:
co_table = Table(rows=co_sources)

co_table

id,ra,dec,z,jup,snr
str21,str11,str12,float64,int64,float64
1,03:32:38.54,-27:46:34.62,2.543,3,37.7
2,03:32:42.38,-27:47:07.92,1.317,2,17.9
3,03:32:41.02,-27:46:31.56,2.453,3,15.8
4,03:32:34.44,-27:46:59.82,1.414,2,15.5
5,03:32:39.76,-27:46:11.58,1.55,2,15.0
6,03:32:39.90,-27:47:15.12,1.095,2,11.9
7,03:32:43.53,-27:46:39.47,2.697,3,10.9
8,03:32:35.58,-27:46:26.16,1.382,2,9.5
9,03:32:44.03,-27:46:36.05,2.698,3,9.3


In [10]:
co_table.write(
    CATALOGS / "aspecs_co_catalog.ecsv",
    overwrite=True
)

In [11]:
# ==========================================================
# Phase 1A
# Load the ASPECS CO Catalog
# ==========================================================

co_table = Table.read(CATALOGS / "aspecs_co_catalog.ecsv")

print(co_table)

 id       ra         dec        z   jup snr 
---- ----------- ------------ ----- --- ----
   1 03:32:38.54 -27:46:34.62 2.543   3 37.7
   2 03:32:42.38 -27:47:07.92 1.317   2 17.9
   3 03:32:41.02 -27:46:31.56 2.453   3 15.8
   4 03:32:34.44 -27:46:59.82 1.414   2 15.5
   5 03:32:39.76 -27:46:11.58  1.55   2 15.0
   6 03:32:39.90 -27:47:15.12 1.095   2 11.9
   7 03:32:43.53 -27:46:39.47 2.697   3 10.9
   8 03:32:35.58 -27:46:26.16 1.382   2  9.5
   9 03:32:44.03 -27:46:36.05 2.698   3  9.3
  10 03:32:42.98 -27:46:50.45 1.037   2  8.7
  11 03:32:39.80 -27:46:53.70 1.096   2  7.9
  12 03:32:36.21 -27:46:27.78 2.574   3  7.0
  13 03:32:35.56 -27:47:04.32 3.601   4  6.8
  14 03:32:34.84 -27:46:40.74 1.098   2  6.7
  15 03:32:36.48 -27:46:31.92 1.096   2  6.5
  16 03:32:39.92 -27:46:07.44 1.294   2  6.4
MP01 03:32:37.30 -27:45:57.80 1.096   2  4.5
MP02 03:32:35.48 -27:46:26.50 1.087   2  4.0


In [12]:
# ==========================================================
# Phase 1B
# Create Sky Coordinates
# ==========================================================

co_coords = SkyCoord(
    co_table["ra"],
    co_table["dec"],
    unit=(u.hourangle, u.deg),
    frame="icrs",
)

print(co_coords[:5])

<SkyCoord (ICRS): (ra, dec) in deg
    [(53.16058333, -27.77628333), (53.17658333, -27.78553333),
     (53.17091667, -27.77543333), (53.1435    , -27.78328333),
     (53.16566667, -27.76988333)]>


In [13]:
# ==========================================================
# Phase 1B
# Compute Angular Separations from C19
# ==========================================================

# Compute separations
co_table["sep_arcsec"] = c19.separation(co_coords).arcsecond

# Sort by increasing separation
co_table.sort("sep_arcsec")

# Display the most relevant columns
co_table["id", "ra", "dec", "z", "sep_arcsec"]

id,ra,dec,z,sep_arcsec
str4,str11,str12,float64,float64
12,03:32:36.21,-27:46:27.78,2.574,0.3447559525144278
15,03:32:36.48,-27:46:31.92,1.096,5.493610615819969
8,03:32:35.58,-27:46:26.16,1.382,8.302300006107911
MP02,03:32:35.48,-27:46:26.50,1.087,9.541661517826702
14,03:32:34.84,-27:46:40.74,1.098,21.984430049035257
1,03:32:38.54,-27:46:34.62,2.543,31.883377582290304
MP01,03:32:37.30,-27:45:57.80,1.096,33.60179696832994
13,03:32:35.56,-27:47:04.32,3.601,37.26991419671708
4,03:32:34.44,-27:46:59.82,1.414,39.39418890701477


In [14]:
# ==========================================================
# Phase 1B
# Nearby CO Sources
# ==========================================================

within10 = co_table[co_table["sep_arcsec"] <= 10]
within20 = co_table[co_table["sep_arcsec"] <= 20]
within30 = co_table[co_table["sep_arcsec"] <= 30]

print(f"Within 10 arcsec : {len(within10)}")
print(f"Within 20 arcsec : {len(within20)}")
print(f"Within 30 arcsec : {len(within30)}")

Within 10 arcsec : 4
Within 20 arcsec : 4
Within 30 arcsec : 5


In [15]:
print("\nWithin 10 arcsec")
print(within10["id", "z", "sep_arcsec"])

print("\nWithin 20 arcsec")
print(within20["id", "z", "sep_arcsec"])

print("\nWithin 30 arcsec")
print(within30["id", "z", "sep_arcsec"])


Within 10 arcsec
 id    z       sep_arcsec    
---- ----- ------------------
  12 2.574 0.3447559525144278
  15 1.096  5.493610615819969
   8 1.382  8.302300006107911
MP02 1.087  9.541661517826702

Within 20 arcsec
 id    z       sep_arcsec    
---- ----- ------------------
  12 2.574 0.3447559525144278
  15 1.096  5.493610615819969
   8 1.382  8.302300006107911
MP02 1.087  9.541661517826702

Within 30 arcsec
 id    z       sep_arcsec    
---- ----- ------------------
  12 2.574 0.3447559525144278
  15 1.096  5.493610615819969
   8 1.382  8.302300006107911
MP02 1.087  9.541661517826702
  14 1.098 21.984430049035257


In [16]:
# ==========================================================
# Phase 1C
# Redshift Differences
# ==========================================================

co_table["delta_z"] = abs(co_table["z"] - c19_redshift)

co_table["likely_companion"] = co_table["delta_z"] < 0.05

co_table["id", "z", "delta_z", "sep_arcsec", "likely_companion"]

id,z,delta_z,sep_arcsec,likely_companion
str4,float64,float64,float64,bool
12,2.574,0.0,0.3447559525144278,True
15,1.096,1.4779999999999998,5.493610615819969,False
8,1.382,1.192,8.302300006107911,False
MP02,1.087,1.4869999999999999,9.541661517826702,False
14,1.098,1.4759999999999998,21.984430049035257,False
1,2.543,0.030999999999999694,31.883377582290304,True
MP01,1.096,1.4779999999999998,33.60179696832994,False
13,3.601,1.0270000000000001,37.26991419671708,False
4,1.414,1.16,39.39418890701477,False


## Phase 1C Summary

The ASPECS CO catalog was searched for neighboring CO detections around
C19.

Within 30 arcseconds, no additional CO source has a redshift consistent
with C19 (Δz < 0.05).

The nearest redshift-consistent source is CO ID 1, located 31.9
arcseconds from C19 with Δz = 0.031.

These results suggest that C19 is not part of a compact CO-selected
pair within the ASPECS sample, although a wider-scale association
cannot yet be excluded.

In [17]:
# ==========================================================
# Phase 1C
# Classify CO Neighbors
# ==========================================================

def classify_neighbor(row):
    if row["sep_arcsec"] <= 10 and row["delta_z"] < 0.05:
        return "Target (C19)"
    elif row["sep_arcsec"] <= 30 and row["delta_z"] < 0.05:
        return "Local Companion"
    elif row["delta_z"] < 0.05:
        return "Candidate Same-Redshift Source"
    else:
        return "Different Redshift"

co_table["association_status"] = [
    classify_neighbor(row) for row in co_table
]

co_table["id", "z", "delta_z", "sep_arcsec", "association_status"]

id,z,delta_z,sep_arcsec,association_status
str4,float64,float64,float64,str30
12,2.574,0.0,0.3447559525144278,Target (C19)
15,1.096,1.4779999999999998,5.493610615819969,Different Redshift
8,1.382,1.192,8.302300006107911,Different Redshift
MP02,1.087,1.4869999999999999,9.541661517826702,Different Redshift
14,1.098,1.4759999999999998,21.984430049035257,Different Redshift
1,2.543,0.030999999999999694,31.883377582290304,Candidate Same-Redshift Source
MP01,1.096,1.4779999999999998,33.60179696832994,Different Redshift
13,3.601,1.0270000000000001,37.26991419671708,Different Redshift
4,1.414,1.16,39.39418890701477,Different Redshift


In [18]:
# ==========================================================
# Phase 1C
# Save CO Neighbor Analysis
# ==========================================================

co_table.write(
    CATALOGS / "obs001_co_neighbor_analysis.ecsv",
    overwrite=True
)

print("Saved:", CATALOGS / "obs001_co_neighbor_analysis.ecsv")

Saved: /home/glenn/Projects/CosmicIntelligenceLab/data/catalogs/obs001_co_neighbor_analysis.ecsv


## Phase 1C Results

The ASPECS CO catalog was searched for neighboring CO detections
around ASPECS C19.

### Measurements

- Four CO detections lie within 10 arcseconds.
- Four CO detections lie within 20 arcseconds.
- Five CO detections lie within 30 arcseconds.

Only the target galaxy (CO ID 12) has Δz = 0.

CO ID 1 has Δz = 0.031 but lies 31.9 arcseconds from C19.

All remaining CO detections have substantially different redshifts.

### Preliminary Interpretation

No additional CO-selected galaxy within 30 arcseconds satisfies the
current same-redshift criterion (Δz < 0.05).

CO ID 1 lies just outside the adopted search radius and is retained
as a candidate for investigation during the projected-separation
analysis.

No conclusions regarding physical association are drawn at this stage.

In [19]:
# ==========================================================
# Phase 1C
# Create Analysis Table
# ==========================================================

analysis_table = co_table.copy()

analysis_table.write(
    CATALOGS / "obs001_co_neighbor_analysis.ecsv",
    overwrite=True
)

print("Saved:", CATALOGS / "obs001_co_neighbor_analysis.ecsv")

Saved: /home/glenn/Projects/CosmicIntelligenceLab/data/catalogs/obs001_co_neighbor_analysis.ecsv


In [20]:
# ==========================================================
# Phase 1D
# Physical Scale at the Redshift of C19
# ==========================================================

scale = Planck18.kpc_proper_per_arcmin(c19_redshift)

print(scale)

492.6769714447917 kpc / arcmin


In [21]:
kpc_per_arcsec = scale / 60

print(kpc_per_arcsec)

8.211282857413195 kpc / arcmin


In [24]:
# ==========================================================
# Phase 1D
# Projected Physical Separations
# ==========================================================

analysis_table["projected_kpc"] = (
    analysis_table["sep_arcsec"] *
    kpc_per_arcsec.value
)

analysis_table["id",
               "z",
               "sep_arcsec",
               "projected_kpc",
               "association_status"]

id,z,sep_arcsec,projected_kpc,association_status
str4,float64,float64,float64,str30
12,2.574,0.3447559525144278,2.830888642872879,Target (C19)
15,1.096,5.493610615819969,45.109590674985654,Different Redshift
8,1.382,8.302300006107911,68.17253371725536,Different Redshift
MP02,1.087,9.541661517826702,78.34928165256957,Different Redshift
14,1.098,21.984430049035257,180.52037359164274,Different Redshift
1,2.543,31.883377582290304,261.80343177789257,Candidate Same-Redshift Source
MP01,1.096,33.60179696832994,275.91385942432635,Different Redshift
13,3.601,37.26991419671708,306.03380754076363,Different Redshift
4,1.414,39.39418890701477,323.47682805386745,Different Redshift


In [25]:
# ==========================================================
# Phase 1D
# Velocity Differences
# ==========================================================

from astropy.constants import c

analysis_table["delta_v_kms"] = (
    c.to("km/s").value *
    analysis_table["delta_z"] /
    (1 + c19_redshift)
)

analysis_table["id",
               "z",
               "delta_z",
               "delta_v_kms",
               "projected_kpc"]

id,z,delta_z,delta_v_kms,projected_kpc
str4,float64,float64,float64,float64
12,2.574,0.0,0.0,2.830888642872879
15,1.096,1.4779999999999998,123976.84748852823,45.109590674985654
8,1.382,1.192,99986.74032904308,68.17253371725536
MP02,1.087,1.4869999999999999,124731.78093060995,78.34928165256957
14,1.098,1.4759999999999998,123809.08450139897,180.52037359164274
1,2.543,0.030999999999999694,2600.3263005036115,261.80343177789257
MP01,1.096,1.4779999999999998,123976.84748852823,275.91385942432635
13,3.601,1.0270000000000001,86146.29389087859,306.03380754076363
4,1.414,1.16,97302.53253497482,323.47682805386745


In [28]:
# ==========================================================
# Phase 2A
# Load the Object 39 Comparison Catalog
# ==========================================================

comparison = Table.read(CATALOGS / "object39_comparison_catalog.ecsv")

print(f"Number of galaxies: {len(comparison)}")
comparison

Number of galaxies: 8


Seq,RAJ2000,DEJ2000,best_z,zspec,zphot,sep_arcsec,AGN,Ksmag,IR36mag,IR45mag
,deg,deg,,,,,,mag,mag,mag
int32,float64,float64,float64,float64,float64,float64,int16,float64,float64,float64
10989,53.160816,-27.775379,0.64,--,0.64,0.2145043835039095,0,20.489,20.838,21.232
10645,53.161629,-27.780247,0.619,0.619,0.62,17.51286173275635,0,19.964,20.638,21.002
10646,53.158283,-27.780985,0.59,--,0.59,21.523854534081554,0,23.842,24.184,24.771
10587,53.158138,-27.781094,0.64,--,0.64,22.061988696899594,0,21.563,22.083,22.362
11316,53.168579,-27.772926,0.61,--,0.61,26.366529365011314,0,25.055,26.199,26.016
11688,53.158356,-27.768261,0.63,--,0.63,26.98575314963197,0,26.093,25.279,25.962
11309,53.151211,-27.772839,0.6,--,0.6,31.95350715576038,0,23.652,24.066,24.096
11209,53.16993,-27.77103,0.67,--,0.67,33.119586168819694,0,19.292,19.5,19.952


In [29]:
# ==========================================================
# Phase 2B
# Compare Redshifts
# ==========================================================

comparison["delta_z"] = abs(comparison["best_z"] - c19_redshift)

comparison.sort("delta_z")

comparison["Seq",
           "best_z",
           "zspec",
           "zphot",
           "sep_arcsec",
           "delta_z"]

Seq,best_z,zspec,zphot,sep_arcsec,delta_z
int32,float64,float64,float64,float64,float64
11209,0.67,--,0.67,33.119586168819694,1.904
10989,0.64,--,0.64,0.2145043835039095,1.9339999999999997
10587,0.64,--,0.64,22.061988696899594,1.9339999999999997
11688,0.63,--,0.63,26.98575314963197,1.944
10645,0.619,0.619,0.62,17.51286173275635,1.9549999999999998
11316,0.61,--,0.61,26.366529365011314,1.964
11309,0.6,--,0.6,31.95350715576038,1.9739999999999998
10646,0.59,--,0.59,21.523854534081554,1.984


## Observation 2 — Nearby Optical Galaxies

The Object 39 comparison catalog contains eight galaxies within
approximately 33 arcseconds of the candidate C19 counterpart.

All eight galaxies have best-fit redshifts between z = 0.59 and
z = 0.67.

No galaxy in this comparison catalog has a redshift consistent with
C19 (z = 2.574).

These nearby galaxies therefore represent unrelated foreground
objects projected along the same line of sight.